### PR TITLE
Stampy redesign

### PR DESCRIPTION
**Pull request by mruwnik**
New chatbot UI deployment.

Testable version at https://redesign.aprillion.workers.dev/chat/

### CODE REVIEW

**Comment by LeMurphant** [comment]
I can't vouch for the code, but the functionality looks good enough. 

The two issues that bug me most are
https://github.com/StampyAI/stampy-ui/issues/736
https://github.com/StampyAI/stampy-ui/issues/643
Still, I think it's ok for a first release.

File: app/root.css [hunk 1]
```diff
@@ -192,145 +207,212 @@ h2 {
 }
 
 .padding-bottom-4 {
-  padding-bottom: var(--spacing-4);
+  padding-bottom: var(--spacing-4) !important;
```
Comment (Aprillion) [inline app/root.css:210]: I don't like overusing important ... but if no one has time to remove the wrong padding from selectors with higher priority that overwrite these classes, it's not a blocker 😭

File: app/root.css [hunk 2]
```diff
@@ -340,7 +422,7 @@ a {
 
 a:hover {
   text-decoration: none;
-  color: var(--colors-teal-500, #1d9089);
+  /* color: var(--colors-teal-500, #1d9089); */
```
Comment (Aprillion) [inline app/root.css:425]: ?

File: app/root.tsx [hunk 3]
```diff
@@ -133,7 +133,7 @@ function Head() {
   return (
     <head>
       <meta charSet="utf-8" />
-      <meta name="viewport" content="width=device-width,initial-scale=1" />
+      <meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
```
Comment (Aprillion) [inline app/root.tsx:136]: huh, why do we need to disable zoom-out on mobile? there is no issue linked in the commit...

File: app/routes/questions.$questionId.$.tsx [hunk 4]
```diff
@@ -132,8 +140,8 @@ export default function RenderArticle() {
             />
           }
         >
-          <Await resolve={Promise.all([question, tags])}>
-            {([resolvedQuestion, resolvedTags]) => {
+          <Await resolve={question}>
```
Comment (Aprillion) [inline app/routes/questions.$questionId.$.tsx:143]: 👍

File: app/routes/questions.actions.tsx [hunk 5]
```diff
@@ -149,20 +163,20 @@ export const Action = ({
     const response = await fetch('/questions/actions', {method: 'POST', body: searchParams})
 
     if (response.ok !== true) {
-      setActionTaken(!actionTaken)
+      // don't ask
+      setActionTaken(actionTaken)
```
Comment (Aprillion) [inline app/routes/questions.actions.tsx:167]: ?!?

File: app/routes/questions.editors.tsx
Comment (Aprillion) [inline app/routes/questions.editors.tsx:1]: why not deleting the file?

File: app/routes/questions.old.$questionId.tsx
Comment (Aprillion) [inline app/routes/questions.old.$questionId.tsx:1]: another non-deleted file?

File: public/favicon-min-512.png
Comment (Aprillion) [inline public/favicon-min-512.png:1]: we deleted the code to switch between 2 favicons => this file shouldn't be re-added

File: public/favicon-min.ico
Comment (Aprillion) [inline public/favicon-min.ico:1]: and this file is not needed either

**Review by Aprillion — DISMISSED** [review]
✅ few comments, but I didn't notice anything that should block the release

File: app/hooks/useChat.ts [hunk 6]
```diff
@@ -0,0 +1,293 @@
+export const CHATBOT_URL = 'https://chat.stampy.ai:8443/chat'
```
[the remark this answers is not in the document]
Comment (mruwnik) [reply app/hooks/useChat.ts:1]: it should. But I gave up after trying various things and nothing working

File: app/routes/questions.actions.tsx (hunk 5 above)
Comment (mruwnik) [reply app/routes/questions.actions.tsx:167]: @zarSou9 explain yourself! For the allseeing gaze of Peter has fallen upon you!

Comment (Aprillion) [reply app/routes/questions.actions.tsx:167]: I might have seen that problem [once or twice](https://stackoverflow.com/a/58877875/1176601)... I think the code is fine for that purpose (using `(v) => !v)` would be wrong if the handler is called multiple times), but the comment could improve:
```suggestion
      // if response is not ok, revert to original state as it was before this handler started
      setActionTaken(actionTaken)
```

**Comment by bryceerobertson** [comment]
I think #643 is sufficiently problematic to be launch-blocking, since it seems to consistently happen at around the 3rd or 4th prompt. Other than that looks great.

**Review by bryceerobertson — APPROVED** [review]

**Review by zarSou9 — APPROVED** [review]
Looking great! Especially since #643 is fixed. 🙏 Nemo

....so who's gonna press it?

**Pull request merged**